slice/internal/controller/workload_controller.go

```diff
@@ -280,27 +379,63 @@ func (r *WorkloadReconciler) updateWorkloadAdmissionCheckStatus(ctx context.Cont
 	return err
 }
 
-// syncAdmissionCheckStatus syncs the admission check status with the state of the slice.
-func (r *WorkloadReconciler) syncAdmissionCheckStatus(ctx context.Context, wl *kueue.Workload, ac *kueue.AdmissionCheckState, slice *v1alpha1.Slice) error {
+func joinSliceNames(slices []v1alpha1.Slice) string {
+	sliceNames := make([]string, len(slices))
+	for index, slice := range slices {
+		sliceNames[index] = fmt.Sprintf("%q", client.ObjectKeyFromObject(&slice))
+	}
+	return strings.Join(sliceNames, ", ")
+}
+
+// syncAdmissionCheckStatus syncs the admission check status with the state of the Slices.
+func (r *WorkloadReconciler) syncAdmissionCheckStatus(ctx context.Context, wl *kueue.Workload, ac *kueue.AdmissionCheckState, slices []v1alpha1.Slice) error {
 	originalState := ac.State
+	originalMessage := ac.Message
+
+	slicesByStatus := make(map[v1alpha1.SliceConditionType][]v1alpha1.Slice)
+	for _, slice := range slices {
+		if core.Forming(&slice) {
+			slicesByStatus[v1alpha1.Forming] = append(slicesByStatus[v1alpha1.Forming], slice)
+		}
+
+		if core.Ready(&slice) {
+			slicesByStatus[v1alpha1.Ready] = append(slicesByStatus[v1alpha1.Ready], slice)
+		}
```

Comment:
```suggestion
		for _, status := range []v1alpha1.SliceConditionType{v1alpha1.Forming, v1alpha1.Ready, v1alpha1.Degraded, v1alpha1.Deformed, v1alpha1.Error} {
			if meta.IsStatusConditionTrue(slice.Status.Conditions, string(status)) {
				slicesByStatus[status] = append(slicesByStatus[status], slice)
			}
		}
```
Not sure if better but how about making a loop over all the possible states?

Reply:
Done